src/java.net.http/share/classes/jdk/internal/net/http/Http2ClientImpl.java

```diff
@@ -228,7 +228,10 @@ void stop() {
             connectionPoolLock.unlock();
         }
         do {
-            connections.values().forEach(this::close);
+            for (var entry : connections.entrySet()) {
+                connections.values().forEach(this::close);
+                connections.remove(entry.getKey(), entry.getValue());
+            }
```

Comment:
I'd suggest to change `this::close(Http2Connection)` to return boolean (instead of void) and make it always return`true`, and then use `removeIf` instead of forEach:

```
connections.values().removeIf(this::close);
```

Reply:
So effect of your suggested changes is to remove the connection if close returns `true` as supposed to just removing all connections and

This change is not entirely related to this PR. I will revert it and create a seperate issue. Though the suggestion is good and I will apply it to the give fix for that issue.

Reply:
https://bugs.openjdk.org/browse/JDK-8318492